File: app/src/main/java_firstpage/com/yongliang/houylPage/section/factory/FragmentSectionBaseFactory.java
package com.yongliang.houylPage.section.factory;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.yongliang.houylPage.section.base.BaseSection;


/**
 * created by houyl
 * on  9:18 AM
 */
public abstract class FragmentSectionBaseFactory {

   public abstract <T extends BaseSection> T createProduct(Class<T> clz, Fragment context, Bundle bundle);
}

File: app/src/main/java/common_weight/recy/demo1/factory/FragmentSectionBaseFactory.kt
package com.txx.app.main.section.factory

import android.os.Bundle
import androidx.fragment.app.Fragment
import common_weight.recy.demo1.baseView.FragmentBaseSectionProduct

/**
 * created by houyl
 * on  9:18 AM
 */
interface FragmentSectionBaseFactory {
     fun <T : FragmentBaseSectionProduct<*, *>?> createProduct(clz: Class<T>, context: Fragment, bundle: Bundle): T
}